### ISSUE TITLE
usages aren't returned if no tiers

### ISSUE THREAD

**Issue by israelidanny** [killbill/killbill#1303]
I have this sample catalog, and even though there are usages limits specified, the API does not return them.

Instead I get if I request the phase (similar results if I look at the subscription or even the JSON version of the catalog):
```
{
  "type": "EVERGREEN",
  "prices": [
    {
      "currency": "USD",
      "value": 0
    }
  ],
  "fixedPrices": [],
  "duration": {
    "unit": "UNLIMITED",
    "number": -1
  },
  "usages": [
    {
      "billingPeriod": "NO_BILLING_PERIOD",
      "tiers": []
    }
  ]
}
```

I would expect usages to be shown there. This would be useful for a use case where I need to have an API subscription with certain limits, and be aware of them when pulling the subscription info out of killbill.

Here is my catalog:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<catalog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="CatalogSchema.xsd">
    <effectiveDate>2020-05-11T13:22:15Z</effectiveDate>
    <catalogName>DEFAULT</catalogName>
    <recurringBillingMode>IN_ADVANCE</recurringBillingMode>    
    <currencies>
        <currency>USD</currency>
    </currencies>
    <units>
        <unit name="cloud-hives"/>
        <unit name="custom-hives"/>
        <unit name="test-interval-minutes"/>
        <unit name="history-days"/>
        <unit name="scouts"/>
    </units>
    <products>
        <product name="Personal" prettyName="Personal">
            <category>BASE</category>
            <included/>
            <available/>
        </product>
        <product name="Advance" prettyName="Advance">
            <category>BASE</category>
            <included>
                <addonProduct>SimpleAlerts</addonProduct>
            </included>
            <available/>
        </product>
        <product name="Professional" prettyName="Professional">
            <category>BASE</category>
            <included>
                <addonProduct>SimpleAlerts</addonProduct>
                <addonProduct>ProactiveAlerts</addonProduct>
            </included>
            <available/>
        </product>
        <product name="Enterprise" prettyName="Enterprise">
            <category>BASE</category>
            <included>
                <addonProduct>SimpleAlerts</addonProduct>
                <addonProduct>ProactiveAlerts</addonProduct>
            </included>
            <available/>
        </product>
        <product name="SimpleAlerts" prettyName="Simple Alerts">
            <category>ADD_ON</category>
        </product>
        <product name="ProactiveAlerts" prettyName="Proactive Alerts">
            <category>ADD_ON</category>
        </product>
    </products>
    <rules>
        <changePolicy>
            <changePolicyCase>
                <policy>IMMEDIATE</policy>
            </changePolicyCase>
        </changePolicy>
        <changeAlignment>
            <changeAlignmentCase>
                <alignment>START_OF_BUNDLE</alignment>
            </changeAlignmentCase>
        </changeAlignment>
        <cancelPolicy>
            <cancelPolicyCase>
                <policy>IMMEDIATE</policy>
            </cancelPolicyCase>
        </cancelPolicy>
        <createAlignment>
            <createAlignmentCase>
                <alignment>START_OF_BUNDLE</alignment>
            </createAlignmentCase>
        </createAlignment>
        <billingAlignment>
            <billingAlignmentCase>
                <alignment>ACCOUNT</alignment>
            </billingAlignmentCase>
        </billingAlignment>
        <priceList>
            <priceListCase>
                <toPriceList>DEFAULT</toPriceList>
            </priceListCase>
        </priceList>
    </rules>
    <plans>
        <plan name="enterprise-yearly" prettyName="Enterprise Yearly">
            <product>Enterprise</product>
            <recurringBillingMode>IN_ADVANCE</recurringBillingMode>
            <initialPhases>
                <phase type="TRIAL">
                    <duration>
                        <unit>DAYS</unit>
                        <number>14</number>
                    </duration>
                    <fixed type="ONE_TIME">
                        <fixedPrice>
                            <price>
                                <currency>USD</currency>
                                <value>0</value>
                            </price>
                        </fixedPrice>
                    </fixed>
                    <usages/>
                </phase>
            </initialPhases>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                    <number>-1</number>
                </duration>
                <recurring>
                    <billingPeriod>ANNUAL</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>0</value>
                        </price>
                    </recurringPrice>
                </recurring>
                <usages/>
            </finalPhase>
            <plansAllowedInBundle>1</plansAllowedInBundle>
        </plan>
        <plan name="enterprise-monthly" prettyName="Enterprise Monthly">
            <product>Enterprise</product>
            <recurringBillingMode>IN_ADVANCE</recurringBillingMode>
            <initialPhases>
                <phase type="TRIAL">
                    <duration>
                        <unit>DAYS</unit>
                        <number>14</number>
                    </duration>
                    <fixed type="ONE_TIME">
                        <fixedPrice>
                            <price>
                                <currency>USD</currency>
                                <value>0</value>
                            </price>
                        </fixedPrice>
                    </fixed>
                    <usages/>
                </phase>
            </initialPhases>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                    <number>-1</number>
                </duration>
                <recurring>
                    <billingPeriod>MONTHLY</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>0</value>
                        </price>
                    </recurringPrice>
                </recurring>
                <usages/>
            </finalPhase>
            <plansAllowedInBundle>1</plansAllowedInBundle>
        </plan>
        <plan name="professional-yearly" prettyName="Professional Yearly">
            <product>Professional</product>
            <recurringBillingMode>IN_ADVANCE</recurringBillingMode>
            <initialPhases>
                <phase type="TRIAL">
                    <duration>
                        <unit>DAYS</unit>
                        <number>14</number>
                    </duration>
                    <fixed type="ONE_TIME">
                        <fixedPrice>
                            <price>
                                <currency>USD</currency>
                                <value>0</value>
                            </price>
                        </fixedPrice>
                    </fixed>
                    <usages/>
                </phase>
            </initialPhases>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                    <number>-1</number>
                </duration>
                <recurring>
                    <billingPeriod>ANNUAL</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>0</value>
                        </price>
                    </recurringPrice>
                </recurring>
                <usages>
                    <usage name="professional-yearly-usage" billingMode="IN_ADVANCE" usageType="CAPACITY">
                        <billingPeriod>NO_BILLING_PERIOD</billingPeriod>
                                <limits>
                                    <limit>
                                       <unit>scouts</unit>
                                       <max>10</max>
                                   </limit>
                                   <limit>
                                       <unit>custom-hives</unit>
                                       <max>4</max>
                                   </limit>
                                   <limit>
                                       <unit>history-days</unit>
                                       <max>30</max>
                                   </limit>
                                   <limit>
                                       <unit>test-interval-minutes</unit>
                                       <min>15</min>
                                   </limit>
                               </limits>
                    </usage>
                </usages>
            </finalPhase>
            <plansAllowedInBundle>1</plansAllowedInBundle>
        </plan>
        <plan name="professional-monthly" prettyName="Professional Monthly">
            <product>Professional</product>
            <recurringBillingMode>IN_ADVANCE</recurringBillingMode>
            <initialPhases>
                <phase type="TRIAL">
                    <duration>
                        <unit>DAYS</unit>
                        <number>14</number>
                    </duration>
                    <fixed type="ONE_TIME">
                        <fixedPrice>
                            <price>
                                <currency>USD</currency>
                                <value>0</value>
                            </price>
                        </fixedPrice>
                    </fixed>
                    <usages/>
                </phase>
            </initialPhases>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                    <number>-1</number>
                </duration>
                <recurring>
                    <billingPeriod>MONTHLY</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>0</value>
                        </price>
                    </recurringPrice>
                </recurring>
                <usages/>
            </finalPhase>
            <plansAllowedInBundle>1</plansAllowedInBundle>
        </plan>
        <plan name="advance-yearly" prettyName="Advance Monthly">
            <product>Advance</product>
            <recurringBillingMode>IN_ADVANCE</recurringBillingMode>
            <initialPhases>
                <phase type="TRIAL">
                    <duration>
                        <unit>DAYS</unit>
                        <number>14</number>
                    </duration>
                    <fixed type="ONE_TIME">
                        <fixedPrice>
                            <price>
                                <currency>USD</currency>
                                <value>0</value>
                            </price>
                        </fixedPrice>
                    </fixed>
                    <usages/>
                </phase>
            </initialPhases>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                    <number>-1</number>
                </duration>
                <recurring>
                    <billingPeriod>ANNUAL</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>0</value>
                        </price>
                    </recurringPrice>
                </recurring>
                <usages/>
            </finalPhase>
            <plansAllowedInBundle>1</plansAllowedInBundle>
        </plan>
        <plan name="advance-monthly" prettyName="Advance Monthly">
            <product>Advance</product>
            <recurringBillingMode>IN_ADVANCE</recurringBillingMode>
            <initialPhases>
                <phase type="TRIAL">
                    <duration>
                        <unit>DAYS</unit>
                        <number>14</number>
                    </duration>
                    <fixed type="ONE_TIME">
                        <fixedPrice>
                            <price>
                                <currency>USD</currency>
                                <value>0</value>
                            </price>
                        </fixedPrice>
                    </fixed>
                    <usages/>
                </phase>
            </initialPhases>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                    <number>-1</number>
                </duration>
                <recurring>
                    <billingPeriod>MONTHLY</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>0</value>
                        </price>
                    </recurringPrice>
                </recurring>
                <usages/>
            </finalPhase>
            <plansAllowedInBundle>1</plansAllowedInBundle>
        </plan>
        <plan name="personal" prettyName="Personal">
            <product>Personal</product>
            <recurringBillingMode>IN_ADVANCE</recurringBillingMode>
            <initialPhases/>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                    <number>-1</number>
                </duration>
                <recurring>
                    <billingPeriod>MONTHLY</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>0</value>
                        </price>
                    </recurringPrice>
                </recurring>
            </finalPhase>
            <plansAllowedInBundle>1</plansAllowedInBundle>
        </plan>
    </plans>
    <priceLists>
        <defaultPriceList name="DEFAULT" prettyName="DEFAULT">
            <plans>
                <plan>personal</plan>
                <plan>enterprise-monthly</plan>
                <plan>professional-monthly</plan>
                <plan>advance-monthly</plan>
                <plan>enterprise-yearly</plan>
                <plan>professional-yearly</plan>
                <plan>advance-yearly</plan>
            </plans>
        </defaultPriceList>
    </priceLists>
</catalog>
```

**Comment by sbrossie** [comment]
Same thread as this
<https://groups.google.com/forum/#!topic/killbilling-users/5TZhPb60Xaw>?

**Comment by israelidanny** [comment]
The only difference I saw between the provided catalog and what I have is that it defines:
`<usage name="professional-yearly-usage" billingMode="IN_ARREAR" usageType="CAPACITY">`

Instead of:
`<usage name="professional-yearly-usage" billingMode="IN_ADVANCE" usageType="CAPACITY">`

But then again at the top it defines:
`<recurringBillingMode>IN_ADVANCE</recurringBillingMode>`

How can those two coexist?

Also, it seems like a bug to me that I can't define tier limits IN_ADVANCE with billing. Those are hard limits, and customers pay in advance for them.

In short, the catalog now compiles, but there seems to really be a logic bug here.

**Comment by sbrossie** [comment]
> Also, it seems like a bug to me that I can't define tier limits IN_ADVANCE with billing. Those are hard limits, and customers pay in advance for them.

We have *only* implemented usage billing `IN_ARREAR`. The `IN_ADVANCE` is a very different model, typically one where customers rely on prepaid credits.


@israelidanny For the future, we typically don't offer support though tickets, please use the mailing list. And we typically create issues ourselves to be able to stay on top of it. Closing ticket.

**Comment by israelidanny** [comment]
Alrighty, thanks for the clarification and I'll use the group exclusively
next time 👌
